src/components/HeaderBack.js

```diff
@@ -1,66 +1,120 @@
-import React from 'react';
-import { 
-  StyleSheet, 
-  Text, 
-  View, 
+import React from "react";
+import {
+  StyleSheet,
+  Text,
+  View,
   Platform,
   TouchableOpacity,
   Image,
-} from 'react-native';
-
-import chevron_left_img from '../assets/images/icon_chevron-left.png';
+  Alert,
+} from "react-native";
 
+import chevron_left_img from "../assets/images/icon_chevron-left.png";
 
 export default class HeaderBack extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  returnToLocationScreen() {
+    this.props.navigation.navigate("Location", {
+      category: this.props.category,
+      location: this.props.location,
+      latitude: this.props.latitude,
+      longitude: this.props.longitude,
+      description: this.props.description,
+      image1: this.props.image1,
+      image2: this.props.image2,
+      firstName: this.props.firstName,
+      lastName: this.props.lastName,
+      email: this.props.email,
+      phone: this.props.phone,
```

Comment:
**Nitpcik**: This group of props is identical to the repeat instances below. If there's a good chance all three instances will remain the same, we can avoid repeating ourselves by creating a function that returns this object in all three places.

Reply:
I'll refactor this component and push the new code.

Reply:
I extracted the redundant code into a navigateBackToPrevious function.